src/templates/Gov/CreateProposal/index.tsx

```diff
@@ -0,0 +1,351 @@
+    setProposalAddress(event.target.value)
+  }
+
+  function handleProposalParamanters(
+    event: React.ChangeEvent<HTMLInputElement>
+  ) {
+    setProposalParamanters(event.target.value)
+  }
+
+  function extractFunctionsList(value: string) {
+    const regex = /(\w+\([^)]*\))/g
+
+    return value.replace(/\s/g, '').match(regex) ?? []
+  }
+
+  function extractParamentersTypeList(value: string) {
+    const regex = /\(([^)]*)\)/g
+
+    const paramentersList = [...value.matchAll(regex)]
+
+    return paramentersList.flatMap(item => item[1].split(','))
+  }
+
+  function checkForumLink(link: string) {
+    const forumLink = 'https://gov.kassandra.finance/'
+
+    return link.startsWith(forumLink)
+  }
+
+  function extractParamantersValueList(value: string) {
+    const regex = /\[.*?\]/g
+
+    const match = value.match(regex)
+    const paramantersValue = match && match.map(item => item.slice(1, -1))
+
+    return paramantersValue
+  }
+
+  async function handleOnSubmit(event: FormEvent<HTMLFormElement>) {
+    event.preventDefault()
+
+    const descriptionWithTitle =
+      '# ' + proposalTitle + '\n' + proposalDescription
+
+    if (proposalForumLink) {
+      const isForumLink = checkForumLink(proposalForumLink)
+
+      if (!isForumLink) {
+        dispatch(
+          setModalAlertText({
+            errorText:
+              'Forum Link Incorrect. Please check the link and try again.'
+          })
+        )
+        return
+      }
+    }
+
+    if (!isOpenAdvancedOptions) {
+      const defaultTargets = [ethers.ZeroAddress]
+      const defaultValues = ['0']
+      const defaultSignatures = ['']
+      const defaultCallDatas = [ethers.ZeroHash]
+      const descriptionWithForumLink =
+        descriptionWithTitle + '\n' + proposalForumLink
+
+      await propose(
+        defaultTargets,
+        defaultValues,
+        defaultSignatures,
+        defaultCallDatas,
+        descriptionWithForumLink
+      )
+
+      return
+    }
+
+    const functionsList = extractFunctionsList(proposalFunction)
+    const parametersValueList = extractParamantersValueList(
+      proposalParamanters.replace(/\s/g, '')
+    )
+
+    const values = new Array(functionsList.length).fill('0')
+    const proposalAddressFormatted = proposalAddress
+      .replace(/\s/g, '')
+      .split(',')
+
+    if (functionsList.length !== proposalAddressFormatted.length) {
+      dispatch(
+        setModalAlertText({
+          errorText: 'Mismatch between input functions and parameters data.'
+        })
+      )
+      return
+    }
+
+    if (functionsList.length !== parametersValueList?.length) {
+      dispatch(
+        setModalAlertText({
+          errorText: 'Mismatch between input functions and parameters data.'
+        })
+      )
+      return
+    }
+
+    const dataHexstring = functionsList.map((item, index) => {
+      const paramentersTypeList = extractParamentersTypeList(item)
```

Comment:
Typo on "parameters"

```suggestion
      const parametersTypeList = extractParamentersTypeList(item)
```